utilities/ner_indexes/ner_indexes.py

```diff
@@ -0,0 +1,126 @@
+import re
+import csv
+import sys
+import json
+import string
+import difflib
+import argparse
+import unicodedata
+from unidecode import unidecode
+from gensim.parsing.preprocessing import preprocess_string, strip_tags, strip_multiple_whitespaces, strip_punctuation
+
+
+def preprocess_primary_name(name):
+    name = re.sub(r'\s\(.*\)', '', name)
+    return name
+
+
+# Used to skip labels that have non-Latin characters
+def check_latin_chars(s):
+    for ch in s:
+        if ch.isalpha():
+            if 'LATIN' not in unicodedata.name(ch):
+                return False
+    return True
+
+
+def preprocess_text(text):
+    custom_filters = [lambda x: x.lower(), strip_multiple_whitespaces]
+    return unidecode(' '.join(preprocess_string(text, custom_filters)))
+
+
+def get_all_names(record):
+    ror_id = record['id']
+    primary_name = preprocess_primary_name(record['name'])
+    aliases = record['aliases']
+    labels = [label['label'] for label in record.get('labels', [])]
+    names = [preprocess_text(name) for name in [
+        primary_name] + aliases + labels if check_latin_chars(name) and len(name) > 5]
+    return names
+
+
+def create_names_dict(data_dump_file):
+    names_dict = {}
+    with open(data_dump_file, 'r+') as f_in:
+        records = json.load(f_in)
+        for record in records:
+            ror_id = record['id']
+            names_dict[ror_id] = get_all_names(record)
+    return names_dict
+
+
+def find_start_stop_index(substring, string):
+    substring_list = preprocess_text(substring).split()
+    string_list = preprocess_text(string).split()
+    word_to_indices_dict = {}
+    current_index = 0
+    for word in string_list:
+        start_index = current_index
+        stop_index = start_index + len(word)
+        if word not in word_to_indices_dict:
+            word_to_indices_dict[word] = []
+        word_to_indices_dict[word].append((start_index, stop_index))
+        # +1 to skip the space after the current word
+        current_index = stop_index + 1
+
+    first_word_indices = word_to_indices_dict.get(substring_list[0], [])
+    last_word_indices = word_to_indices_dict.get(substring_list[-1], [])
+
+    # Find the shortest substring that contains all words in substring_list
+    shortest_substring_info = None
+    for start_char_index, _ in first_word_indices:
+        for _, stop_char_index in last_word_indices:
+            if stop_char_index >= start_char_index:  # Ensuring we are not looking backwards
+                possible_substring = string[start_char_index:stop_char_index]
+                possible_substring_list = preprocess_text(
```

Comment:
Same question as above, is It necessary to run `preprocess_text` here?

Reply:
Removed.